src/simpleppl.jl

```
@@ -0,0 +1,213 @@
+               y = (0.0, (:μ, :s2), (μ, s2) -> MvNormal(μ, sqrt(s2)), :Stochastic)
+           )
+(s2 = (0.0, (), var"#33#36"(), :Stochastic), μ = (1.0, (), var"#34#37"(), :Logical), y = (0.0, (:μ, :s2), var"#35#38"(), :Stochastic))
+
+julia> Model(nt)
+Model(AbstractPPL.ModelState{(:s2, :μ, :y)}((s2 = 0.0, μ = 1.0, y = 0.0), (s2 = (), μ = (), y = (:μ, :s2)), (s2 = var"#33#36"(), μ = var"#34#37"(), y = var"#35#38"()), (s2 = :Stochastic, μ = :Logical, y = :Stochastic)), AbstractPPL.DAG(
+  ⋅    ⋅    ⋅ 
+  ⋅    ⋅    ⋅ 
+ 1.0  1.0   ⋅ , [2, 1, 3]))
+"""
+struct Model
+    ModelState::ModelState
+    DAG::DAG
+
+    Model(value, input, eval, kind) = new(ModelState(value, input, eval, kind), DAG(input))
+end
+
+@generated function Model(nt::NamedTuple{T}) where T
+    values = [:(nt[$i][$j]) for i in 1:length(T), j in 1:4]
+    m = [:(NamedTuple{T}(($(values[:,i]...), ))) for i in 1:4]
+    return :(Model(($(m...),)...))
+end
+
+nodes(m::Model) = m.DAG.sorted_vertex_list
+
+function Base.show(io::IO, m::Model)
+    print(io, "Nodes: \n")
+    for node in nodes(m)
+        print(io, "$node = ", m[VarName{node}()], "\n")
+    end
+    print(io, "DAG: \n") 
+    display(m.DAG.A)
+end
+
+
+"""
+    adjacency_matrix(inputs::NamedTuple)
+
+For a NamedTuple{T} with vertices `T` paired with tuples of input nodes,
+`adjacency_matrix` constructs the adjacency matrix using the order 
+of variables given by `T`. 
+
+# Examples
+
+julia> inputs = (a = (), b = (), c = (:a, :b))
+(a = (), b = (), c = (:a, :b))
+
+julia> AbstractPPL.adjacency_matrix(inputs)
+3×3 SparseMatrixCSC{Float64, Int64} with 2 stored entries:
+  ⋅    ⋅    ⋅ 
+  ⋅    ⋅    ⋅ 
+ 1.0  1.0   ⋅ 
+"""
+function adjacency_matrix(inputs::NamedTuple)
+    N = length(inputs)
+    nodes = keys(inputs)
+    A = spzeros(N,N)
+    for (i, node) in enumerate(nodes)
+        v_inputs = inputs[node] # try as vector 
+       if v_inputs != () 
+            for inp in v_inputs
+                ind = findall(x -> x == inp, nodes)
```

```suggestion
                ind = findfirst(==(inp), nodes)
```

`findfirst` gives you the first index, or `nothing` in case there is none, which should raise an "unexpected" error.

So somewhere before, there should be a check of the `inputs` for consistency (i.e., all parents exist as keys).